auto_round/autoround.py

```diff
@@ -914,8 +914,8 @@ def quant_layer(self, layer_name, inputs, q_inputs=None, device=torch.device("cp
             train_bs = self.train_bs
             for i in range(0, self.n_samples, train_bs):
                 end_index = min(self.n_samples, i + train_bs)
-                tmp_inputs = inputs[i:end_index, ...].to(device)
-                tmp_output = layer.forward(tmp_inputs).to(cache_device)
+                tmp_input = inputs[i:end_index, ...].to(device)
```

Comment:
this could not reduce cpu memory, move the logic to the for loop in the following, add compute the fp output  each iteration.